website/source/docs/internals/discovery-chain.html.md

```diff
@@ -0,0 +1,217 @@
+  subset](/docs/agent/config-entries/service-resolver.html#defaultsubset)
+  defined then all references made to the original service in any other
+  configuration entry that did not specify a subset will be replaced with the
+  default.
+
+* If a [`service-splitter`](/docs/agent/config-entries/service-splitter.html)
+  is created with a [service
+  split](/docs/agent/config-entries/service-splitter.html#splits) that has its
+  own `service-splitter` then the overall effect is flattened and only a single 
+  aggregate traffic split is ultimately configured in the proxy.
+
+* [`service-resolver`](/docs/agent/config-entries/service-resolver.html)
+  redirect loops must be rejected as invalid.
+
+* [`service-router`](/docs/agent/config-entries/service-router.html) and
+  [`service-splitter`](/docs/agent/config-entries/service-splitter.html)
+  configuration entries require an L7 compatible protocol be set for the
+  service via either a
+  [`service-defaults`](/docs/agent/config-entries/service-defaults.html) or
+  [`proxy-defaults`](/docs/agent/config-entries/proxy-defaults.html) config
+  entry. Violations must be rejected as invalid.
+
+* If an [upstream
+  configuration](/docs/connect/registration/service-registration.html#upstream-configuration-reference)
+  [`datacenter`](/docs/connect/registration/service-registration.html#datacenter)
+  parameter is defined then any configuration entry that does not explicitly
+  refer to a desired datacenter should use that value from the upstream.
+
+## Compilation
+
+To correctly interpret a collection of configuration entries as a valid
+discovery chain we first compile them into a form more directly usable by the
+layers responsible for configuration connect sidecar proxies.
+
+You can interact with the compiler directly using the [discovery-chain
+API](/api/discovery-chain.html).
+
+### Compilation Parameters
+
+* **Service Name** - The service being discovered by name.
+* **Datacenter** - The datacenter to use as the basis of compilation. 
+* **Overrides** - Discovery-time tweaks to apply when compiling. These should
+  be derived from either the
+  [proxy](/docs/connect/registration/service-registration.html#complete-configuration-example)
+  or
+  [upstream](/docs/connect/registration/service-registration.html#upstream-configuration-reference)
+  configurations if either are set.
+
+### Compilation Results
+
+The response is a single wrapped `CompiledDiscoveryChain` field:
+
+```json
+{
+    "Chain": {...<CompiledDiscoveryChain>...}
+}
+```
+
+#### `CompiledDiscoveryChain`
+
+The chain encodes a digraph of [nodes](#discoverygraphnode) and
+[targets](#discoverytarget). Nodes are the compiled representation of various
+discovery chain stages and targets are instructions on how to use the [health
+API](/api/health.html#list-nodes-for-connect-capable-service) to retrieve
+relevant service instance lists.
+
+You should traverse the nodes starting with [`StartNode`](#startnode). The
```

Comment:
I don't see any anchor or heading matching `#startnode`. Same for others below - did you intent to add those around the parameters below?

Reply:
Lists with the leading term wrapped by backticks (like the consul agent options page) will link each term.